File: fixtures/dart_output/put_xput.dart
import 'package:http/http.dart' as http;

void main() async {
  var headers = {
    'Content-Type': 'application/json',
  };

  var data = '{"properties": {"email": {"type": "keyword"}}}';

  var res = await http.put('http://localhost:9200/twitter/_mapping/user?pretty', headers: headers, body: data);
  if (res.statusCode != 200) throw Exception('put error: statusCode= ${res.statusCode}');
  print(res.body);
}

File: test/fixtures/ruby-httparty/put_xput.rb
require 'httparty'
require 'json'

url = 'http://localhost:28139/twitter/_mapping/user?pretty'
headers = {
  'Content-Type': 'application/json',
}
# The object won't be serialized exactly like this
# body = '{"properties": {"email": {"type": "keyword"}}}'
body = {
  'properties' => {
    'email' => {
      'type' => 'keyword'
    }
  }
}.to_json
res = HTTParty.put(url, headers: headers, body: body)
